File: Cosmos-IOS/Cosmostation/Cosmostation/Model/Vote.swift
//
//  IrisVote.swift
//  Cosmostation
//
//  Created by yongjoo on 10/12/2019.
//  Copyright © 2019 wannabit. All rights reserved.
//

import Foundation


public class Vote {
    
    public static let OPTION_YES           = "Yes"
    public static let OPTION_NO            = "No"
    public static let OPTION_VETO          = "NoWithVeto"
    public static let OPTION_ABSTAIN       = "Abstain"
    
    var voter: String = ""
    var proposal_id: String = ""
    var option: String = ""
    
    init() {}
    
    init(_ dictionary: [String: Any]) {
        self.voter = dictionary["voter"] as? String ?? ""
        self.proposal_id = dictionary["proposal_id"] as? String ?? ""
        self.option = dictionary["option"] as? String ?? ""
    }
}

File: app/src/main/java/wannabit/io/cosmostaion/model/type/Vote.java
package wannabit.io.cosmostaion.model.type;

import com.google.gson.annotations.SerializedName;

public class Vote {

    public final static String OPTION_YES           = "Yes";
    public final static String OPTION_NO            = "No";
    public final static String OPTION_VETO          = "NoWithVeto";
    public final static String OPTION_ABSTAIN       = "Abstain";

    @SerializedName("voter")
    public String voter;

    @SerializedName("proposal_id")
    public String proposal_id;

    @SerializedName("option")
    public String option;

    @SerializedName("tx_hash")
    public String tx_hash;

    @SerializedName("timestamp")
    public String timestamp;

    @SerializedName("answer")
    public String answer;

}
